compiler/z/codegen/S390Instruction.cpp

```diff
@@ -1806,6 +1806,35 @@ TR::S390RILInstruction::adjustCallOffsetWithTrampoline(int32_t offset, uint8_t *
    return offsetHalfWords;
    }
 
+void
+TR::S390RILInstruction::addMetaDataForCodeAddress(uint8_t *cursor)
+   {
+   uintptr_t immediateAsAddress = (uintptr_t) (uint32_t) getSourceImmediate();
```

Comment:
We should use `static_cast` here to type cast return value of `getSourceImmediate()`. Also in most the function, we should avoid direct castings and use `static_cast`[1] or `reinterpret_cast`[2] where suitable. 

[1]. https://en.cppreference.com/w/cpp/language/static_cast
[2]. https://en.cppreference.com/w/cpp/language/reinterpret_cast